src/main/java/org/jabref/gui/maintable/BibEntryTableViewModel.java

```diff
@@ -53,6 +60,18 @@ public BibEntryTableViewModel(BibEntry entry, BibDatabaseContext bibDatabaseCont
         this.linkedIdentifiers = createLinkedIdentifiersBinding(entry);
         this.matchedGroups = createMatchedGroupsBinding(bibDatabaseContext, entry);
         this.bibDatabaseContext = bibDatabaseContext;
+
+        searchRankProperty.bind(Bindings.createIntegerBinding(() -> {
+            if (matchedBySearchProperty.get() && matchedByGroupProperty.get()) {
+                return 1;
```

Comment:
Can you put these magic numbers into some constants? .e.g First rank, second rank etc (makes it a bit easier to grasp what the numbers mean